modules/custom/activity_viewer/src/Plugin/views/row/ActivityRow.php

```diff
@@ -30,8 +31,32 @@ public function preRender($result) {
 
       foreach ($result as $row) {
         $render_result = [];
-        $render_result[] = $row;
         $entity = $row->_entity;
+        $target_type = $entity->get('field_activity_entity')->getValue()[0]["target_type"];
+
+        if ($target_type === "comment") {
+          $comment_id = $entity->get('field_activity_entity')->getValue()[0]["target_id"];
+          $query = \Drupal::database()->select('comment_field_data', 'c');
+          $query->addField('c', 'entity_id');
+          $query->condition('c.cid', $comment_id);
+          $post_id = $query->execute()->fetchField();
+
+          $query = \Drupal::database()->select('post_field_data', 'p');
+          $query->fields('p');
+          $query->condition('p.id', $post_id);
+          $post = $query->execute()->fetchAll();
```

Comment:
I think it would be better to use Drupals entity query api ( https://www.drupal.org/project/drupal/issues/2068325 ) 
so we can take advantage of entity caching and lazy loading instead of two separate calls to the database just to get the post.